library/src/main/java/com/wdullaer/materialdatetimepicker/time/TimePickerDialog.java

```diff
@@ -801,17 +839,79 @@ public void onValueSelected(Timepoint newValue) {
     }
 
     @Override
-    public void advancePicker(int index) {
-        if(!mAllowAutoAdvance) return;
-        if(index == HOUR_INDEX && mEnableMinutes) {
-            setCurrentItemShowing(MINUTE_INDEX, true, true, false);
+    public void advancePicker(int index, boolean force) {
+        if(!force && !mAllowAutoAdvance) return;
+        if(index == HOUR_INDEX) {
+            if (mEnableMinutes) {
+                setCurrentItemShowing(MINUTE_INDEX, true, true, false);
 
-            String announcement = mSelectHours + ". " + mTimePicker.getMinutes();
-            Utils.tryAccessibilityAnnounce(mTimePicker, announcement);
-        } else if(index == MINUTE_INDEX && mEnableSeconds) {
-            setCurrentItemShowing(SECOND_INDEX, true, true, false);
+                String announcement = mSelectHours + ":" + mTimePicker.getMinutes() + "." + mTimePicker.getSeconds();
+                Utils.tryAccessibilityAnnounce(mTimePicker, announcement);
+            } else {
+                setButtonsFocusable(true);
+            }
+        } else if(index == MINUTE_INDEX ) {
+            if (mEnableSeconds) {
+                setCurrentItemShowing(SECOND_INDEX, true, true, false);
 
-            String announcement = mSelectMinutes+". " + mTimePicker.getSeconds();
+                String announcement = mSelectHours + ":" + mTimePicker.getMinutes() + "." + mTimePicker.getSeconds();
+                Utils.tryAccessibilityAnnounce(mTimePicker, announcement);
+            } else if (Utils.isTv(getActivity().getApplicationContext())) {
+                if (!mIs24HourMode) {
+                    // TODO : check this
+                    setCurrentItemShowing(AM_PM_INDEX, true, false, false);
+                } else {
+                    setButtonsFocusable(true);
+                }
+            }
+        } else if (index == SECOND_INDEX) {
+            if (!mIs24HourMode) {
+                // TODO : check this
```

Comment:
Did you try the sample app on an Android TV box? This should allow you to try out this test case
(it's what I hope to do this afternoon)